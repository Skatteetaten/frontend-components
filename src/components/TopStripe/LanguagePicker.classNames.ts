import { mergeStyleSets } from '@uifabric/merge-styles';
import { FontSizes } from '../utils/fonts';

export const getClassNames = () => {
  return mergeStyleSets({
    languageButton: {
      padding: '4px 16px 0 24px',
      width: '100%',
      selectors: {
        '@media (max-width: 900px)': {
          paddingLeft: 'calc(50% - 48px)'
        }
      }
    },
    checkIcon: {
      display: 'inline',
      position: 'absolute',
      padding: '0 4px',
      marginLeft: '-24px',
      selectors: {
        '@media (max-width: 900px)': {
          left: 'calc(50% - 84px)',
          width: '14px',
          margin: '0 8px',
          fontSize: FontSizes.medium + ' !important'
        }
      }
    },
    flag: {
      marginRight: '4px',
      verticalAlign: 'middle'
    },
    text: {
      selectors: {
        '@media (max-width: 900px)': {
          marginLeft: '4px',
          textDecoration: 'underline'
        }
      }
    }
  });
};
