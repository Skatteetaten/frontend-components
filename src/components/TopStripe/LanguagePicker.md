** Språkvelger i publikumsløsninger **

For å velge mellom norsk bokmål, nynorsk, englesk og samisk i publikumsløsninger.

```js
import { useState } from 'react';
import {
  TopStripe,
  LanguagePicker,
  TopStripeMenu,
  TopStripeButton,
  TopBanner,
  Link
} from '@skatteetaten/frontend-components';

const [language, setLanguage] = useState('nb');

<div>
  <TopStripe>
    <Link path={'#main-content-id'} text={'Hopp til hovedinnhold'} skipLink />
    <Link
      path={'https://www.skatteetaten.no/kontakt/'}
      text={'Kontakt oss'}
      placement="before"
    />
    <TopStripeMenu
      closeMenuAriaLabel="Lukk endre skriftstørrelse"
      title={'Endre skriftstørrelse'}
      showOnMobile={false}
      contentIsMenu={false}
    >
      <div style={{ fontSize: '24px', marginTop: '8px' }}>
        Hold Ctrl-tasten nede (Cmd-tasten på Mac). Trykk på + for å forstørre
        eller - for å forminske.
      </div>
    </TopStripeMenu>
    <LanguagePicker
      selectedLanguage={language}
      setLanguage={setLanguage}
      showOnMobile={true}
      showSami={true}
    />
    <Link path={'#link'} text={'Logg inn'} placement="before" />
  </TopStripe>
</div>;
```

```js noeditor beskrivelse
import LinkGroup from '@skatteetaten/frontend-components/LinkGroup';
const links = [
  {
    text: 'TopBanner (Topp)',
    path: '#topbanner'
  },
  {
    text: 'TopStripe (Toppstripe)',
    path: '#topstripe'
  }
];

<>
  <h3>Språk</h3>
  <p>
    For at folk skal kunne bruke løsningene våre, må de være i et språk de
    forstår. I praksis betyr dette at alle løsninger for publikum skal kunne
    presenteres på norsk bokmål, nynorsk og engelsk. Innhold som er rettet mot
    eller av særlig interesse for den samiske befolkningen skal oversettes til
    samisk.
  </p>

  <h3>Huske språk på tvers</h3>
  <p>
    Vi ønsker å få til en konsistent brukeropplevelse på tvers av løsningene
    våre. Dette gjelder også språket. Som hovedregel skal samme språk brukes på
    tvers i brukerreisen. I praksis betyr det at når en bruker velger engelsk på
    skatteetaten.no, og logger inn, så skal også "Min side" og deretter
    skattemeldingen presentres på engelsk.
  </p>
  <p>
    Dersom du ikke vet hvor brukeren kommer fra, og hvilket språk som ble brukt
    der bør du forsøke å hente brukeren foretrukkede språk fra profil eller
    nettleser før du bruker standardverdien som er norsk bokmål.
  </p>

  <h3>Relaterte komponenter</h3>
  <LinkGroup links={links} />
</>;
```

```js noeditor uu
<>
  <h3>Tips</h3>
  <p>Denne seksjonen en foreløpig tom.</p>

  <h3>Mest relevante WCAG-krav</h3>
  <p>Denne seksjonen en foreløpig tom.</p>

  <h3>WAI-ARIA</h3>
  <p>Denne seksjonen en foreløpig tom.</p>
</>
```
