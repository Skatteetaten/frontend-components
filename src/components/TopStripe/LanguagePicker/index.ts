export { LanguagePicker as default } from './LanguagePicker';
export * from './LanguagePicker';
