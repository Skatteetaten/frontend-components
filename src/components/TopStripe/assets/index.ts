export { default as EnglishFlag } from './EnglishFlag.svg';
export { default as NorwegianFlag } from './NorwegianFlag.svg';
export { default as SamiFlag } from './SamiFlag.svg';
