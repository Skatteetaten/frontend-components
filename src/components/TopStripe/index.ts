export { default } from './TopStripe';
export * from './TopStripeZoom';
export * from './TopStripeMenu';
export * from './TopStripeLanguageSelector';
export * from './TopStripe';
