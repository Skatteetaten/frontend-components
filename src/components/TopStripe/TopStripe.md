** TopStripe er en svart menystripe øverst på innloggede sider for publikum. **

Ikke innlogget ennå:

```js
import TopStripe, {
  TopStripeMenu,
  TopStripeButton
} from '@skatteetaten/frontend-components/TopStripe';
import TopBanner from '@skatteetaten/frontend-components/TopBanner';
import Link from '@skatteetaten/frontend-components/Link';

<div>
  <TopStripe>
    <Link path={'#topstripe'} text={'Kontakt oss'} placement="before" />

    <TopStripeMenu title={'Endre skriftstørrelse'}>
      <div style={{ fontSize: '20px' }}>
        Hold Ctrl-tasten nede (Cmd-tasten på Mac). Trykk på + for å forstørre
        eller - for å forminske.
      </div>
    </TopStripeMenu>
    <TopStripeMenu title={'Language / Språk'}>
      <TopStripeButton ariaLabel={'Norsk'} onClick={() => console.log('NB')}>
        Norsk
      </TopStripeButton>
      <TopStripeButton
        icon={'check'}
        ariaLabel={'Nynorsk'}
        onClick={() => console.log('NN')}
      >
        Nynorsk
      </TopStripeButton>
      <TopStripeButton ariaLabel={'Engelsk'} onClick={() => console.log('EN')}>
        Engelsk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Sørsamisk'}
        onClick={() => console.log('SMA')}
      >
        Sørsamisk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Nordsamisk'}
        onClick={() => console.log('SME')}
      >
        Nordsamisk
      </TopStripeButton>
    </TopStripeMenu>
    <Link path={'#link'} text={'Logg inn'} placement="before" />
  </TopStripe>
  <TopBanner
    external
    title={'Side for publikum'}
    homeText={'Tilbake til skatteetaten.no'}
  />
</div>;
```

Innlogget:

```js
import TopStripe, {
  TopStripeMenu,
  TopStripeButton
} from '@skatteetaten/frontend-components/TopStripe';
import TopBanner from '@skatteetaten/frontend-components/TopBanner';
import Link from '@skatteetaten/frontend-components/Link';
import Icon from '@skatteetaten/frontend-components/Icon';

<div>
  <TopStripe>
    <Link path={'#topstripe'} text={'Kontakt oss'} placement="before" />

    <TopStripeMenu title={'Endre skriftstørrelse'}>
      <div style={{ fontSize: '20px' }}>
        Hold Ctrl-tasten nede (Cmd-tasten på Mac). Trykk på + for å forstørre
        eller - for å forminske.
      </div>
    </TopStripeMenu>
    <TopStripeMenu title={'Language / Språk'}>
      <TopStripeButton ariaLabel={'Norsk'} onClick={() => console.log('NB')}>
        Norsk
      </TopStripeButton>
      <TopStripeButton
        icon={'check'}
        ariaLabel={'Nynorsk'}
        onClick={() => console.log('NN')}
      >
        Nynorsk
      </TopStripeButton>
      <TopStripeButton ariaLabel={'Engelsk'} onClick={() => console.log('EN')}>
        Engelsk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Sørsamisk'}
        onClick={() => console.log('SMA')}
      >
        Sørsamisk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Nordsamisk'}
        onClick={() => console.log('SME')}
      >
        Nordsamisk
      </TopStripeButton>
    </TopStripeMenu>

    <span>
      <Icon iconName="person" />
      Vegard Sandli
    </span>

    <Link path={'#topstripe'} text={'Logg ut'} placement="before" />
  </TopStripe>
  <TopBanner
    external
    title={'Side for publikum'}
    homeText={'Tilbake til skatteetaten.no'}
  />
</div>;
```

Innlogget på mobil:

```js
import TopStripe, {
  TopStripeMenu,
  TopStripeButton
} from '@skatteetaten/frontend-components/TopStripe';
import TopBanner from '@skatteetaten/frontend-components/TopBanner';
import Link from '@skatteetaten/frontend-components/Link';

<div>
  <TopStripe>
    <TopStripeMenu title={'Endre skriftstørrelse'}>
      <div style={{ fontSize: '20px' }}>
        Hold Ctrl-tasten nede (Cmd-tasten på Mac). Trykk på + for å forstørre
        eller - for å forminske.
      </div>
    </TopStripeMenu>
    <TopStripeMenu title={'Language / Språk'}>
      <TopStripeButton ariaLabel={'Norsk'} onClick={() => console.log('NB')}>
        Norsk
      </TopStripeButton>
      <TopStripeButton
        icon={'check'}
        ariaLabel={'Nynorsk'}
        onClick={() => console.log('NN')}
      >
        Nynorsk
      </TopStripeButton>
      <TopStripeButton ariaLabel={'Engelsk'} onClick={() => console.log('EN')}>
        Engelsk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Sørsamisk'}
        onClick={() => console.log('SMA')}
      >
        Sørsamisk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Nordsamisk'}
        onClick={() => console.log('SME')}
      >
        Nordsamisk
      </TopStripeButton>
    </TopStripeMenu>

    <Link path={'#topstripe'} text={'Logg ut'} placement="before" />
  </TopStripe>
  <TopBanner
    external
    title={'Side for publikum'}
    homeText={'Tilbake til skatteetaten.no'}
  />
</div>;
```

```js noeditor beskrivelse
  <h3>Vise innlogging</h3>
  <p>
    TopStripe er den svarte, horisontale stripen helt i toppen. Dette er etter
    DIFIs anbefaling om en felles markering av innloggede tjenester som
    benytter MinID for innlogging. Alle innloggede publikumsløsninger skal ha
    en slik TopStripe. I utgangspunktet skal TopStripe være identisk på tvers av løsningene våre, slik at brukerne kjenner den igjen.
  </p>
  <h3>Overordnede lenker og funksjoner</h3>
  <p>
    TopStripe er også stedet å legge overordnede lenker og funksjoner for
    løsningen, for eksempel «Kontakt oss», «Språk» og «Endre skriftstørrelse». På mobil/nettbrett kan det være aktuelt å skjule enkelte av funksjonene eller legge lenker i en meny.
  </p>



```
