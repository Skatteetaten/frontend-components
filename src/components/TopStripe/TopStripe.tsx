import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Link } from 'office-ui-fabric-react/lib-commonjs/Link';

import ActionButton from '../ActionButton/ActionButton';

import { getClassNames } from './TopStripe.classNames';

export class TopStripe extends React.PureComponent {
  static propTypes = {
    /** Meny elementer */
    items: PropTypes.arrayOf(
      PropTypes.shape({
        /** Intern nøkkel/id i menylisten */
        key: PropTypes.string.isRequired,
        /** Tekst som vises for menyvalget */
        label: PropTypes.string.isRequired,
        /** Ikke vis label, men den blir med som ariaLabel */
        noLabelText: PropTypes.bool,
        /** Icon som vises for menyvalget */
        icon: PropTypes.string,
        /** URL menyvalget skal sende bruker til */
        href: PropTypes.string,
        /** Tekst som skal vises når menyvalget velges */
        info: PropTypes.string,
        /** Undermenyliste */
        items: PropTypes.arrayOf(
          PropTypes.shape({
            /** Intern nøkkel/id i undermenylisten */
            key: PropTypes.string.isRequired,
            /** Tekst som vises for menyvalget */
            label: PropTypes.string.isRequired,
            /** Icon som vises for menyvalget */
            icon: PropTypes.string,
            /** URL menyvalget skal sende bruker til */
            href: PropTypes.string,
            /** Javascript metode som skal kjøres når menyvalget velges */
            onClick: PropTypes.func,
            /** Markering av menyvalget for å indikere at det er valgt */
            selected: PropTypes.bool
          })
        ),
        /** Brukes til å indikere at menyvalget ikke skal vises på små enheter */
        notSmallDevice: PropTypes.bool
      })
    ).isRequired,
    /** Overstyring av "lukk meny"-tekst. Brukes typisk til språkstøtte */
    closeMenuText: PropTypes.string
  };

  static defaultProps = {
    closeMenuText: 'Lukk meny'
  };

  state = {
    activeMenu: null
  };
  // @ts-ignore TODO
  styles = getClassNames(this.props);
  // @ts-ignore TODO
  selectMenu(id) {
    this.setState({ activeMenu: id });
  }

  closeMenu() {
    this.setState({ activeMenu: null });
  }
  // @ts-ignore TODO
  buildDropdownInfo(key, info, show) {
    const classes = [
      this.styles.dropdownContainer,
      this.styles.dropdownContainerRight
    ];

    if (show) {
      classes.push(this.styles.dropdownShow);
    }

    return (
      <div className={classnames(classes)} onClick={() => this.closeMenu()}>
        <div className={this.styles.info}>{info}</div>
        <div className={this.styles.infoClose}>
          <ActionButton
            id={key + '-choice'}
            icon={'ChevronUp'}
            // @ts-ignore TODO
            ariaLabel={this.props.closeMenuText}
          />
          <span className={this.styles.hiddenText}>
            {
              // @ts-ignore TODO
              this.props.closeMenuText
            }
          </span>
        </div>
      </div>
    );
  }
  // @ts-ignore TODO
  buildDropdownMenuLinkChoice(children, href) {
    return (
      <Link
        href={href}
        className={this.styles.dropdownMenuchoiceLink}
        tabIndex={-1}
      >
        {children}
      </Link>
    );
  }
  // @ts-ignore TODO
  buildDropdownMenuChoice(choice, context) {
    const liClasses = [this.styles.dropdownMenuchoice];
    const buttonClasses = [this.styles.dropdownMenuchoiceButton];
    let icon = choice.icon;

    if (choice.selected !== undefined && choice.selected === true) {
      liClasses.push(this.styles.dropdownMenuchoiceActive);
      buttonClasses.push(this.styles.dropdownMenuchoiceButtonActive);

      if (!icon) {
        icon = 'Check';
      }
    }

    if (!icon) {
      buttonClasses.push(this.styles.dropdownMenuchoiceButtonEmptyIcon);
    }

    let clikkFunction = () => {
      choice.onClick();
      context.closeMenu();
    };

    if (choice.href) {
      // @ts-ignore TODO
      clikkFunction = undefined;
    }

    let buttonElement = (
      <ActionButton
        ariaLabel={choice.label}
        icon={icon}
        className={classnames(buttonClasses)}
        onClick={clikkFunction}
      >
        {choice.label}
      </ActionButton>
    );

    if (choice.href) {
      buttonElement = this.buildDropdownMenuLinkChoice(
        buttonElement,
        choice.href
      );
    }

    return (
      <li className={classnames(liClasses)} key={choice.key}>
        {buttonElement}
      </li>
    );
  }
  // @ts-ignore TODO
  buildDropdownMenuCloseChoice(key, closeLabel, context) {
    return (
      <li className={this.styles.dropdownMenuchoiceClose} key={key + '-close'}>
        <ActionButton
          id={key + '-close'}
          icon={'ChevronUp'}
          onClick={() => context.closeMenu()}
          ariaLabel={closeLabel}
        />
        <span className={this.styles.hiddenText}>{closeLabel}</span>
      </li>
    );
  }
  // @ts-ignore TODO
  buildDropdownMenu(key, menuitems, show, rightFixed) {
    const classes = [
      this.styles.dropdownContainer,
      this.styles.dropdownContainerList
    ];

    if (show) {
      classes.push(this.styles.dropdownShow);
    }

    if (rightFixed) {
      classes.push(this.styles.dropdownContainerRight);
    }
    // @ts-ignore TODO
    let menuchoice = menuitems.map(valg =>
      this.buildDropdownMenuChoice(valg, this)
    );

    menuchoice.push(
      // @ts-ignore TODO
      this.buildDropdownMenuCloseChoice(key, this.props.closeMenuText, this)
    );

    return (
      <div className={classnames(classes)}>
        <ul className={this.styles.dropdownList}>{menuchoice}</ul>
      </div>
    );
  }
  // @ts-ignore TODO
  buidNavMenuLinkChoice(children, href) {
    return (
      <Link href={href} className={this.styles.navMenuchoiceLink} tabIndex={-1}>
        {children}
      </Link>
    );
  }

  buildNavMenu() {
    // @ts-ignore TODO
    const itemsLength = this.props.items.length;
    // @ts-ignore TODO
    return this.props.items.map((valg, idx) => {
      const menuChoiceIsActive = this.state.activeMenu === valg.key;
      const classes = [this.styles.navMenuchoice];
      let dropdown = undefined;
      let click = undefined;
      let choiceContent = valg.noLabel ? null : valg.label;

      if (menuChoiceIsActive) {
        classes.push(this.styles.navMenuchoiceActive);
      }

      if (valg.notSmallDevice) {
        classes.push(this.styles.navMenuchoiceNotSmallDevice);
      }

      if (valg.info) {
        dropdown = this.buildDropdownInfo(
          valg.key,
          valg.info,
          menuChoiceIsActive
        );
        click = () => this.selectMenu(valg.key);
      } else if (valg.items) {
        dropdown = this.buildDropdownMenu(
          valg.key,
          valg.items,
          menuChoiceIsActive,
          idx >= itemsLength - 1
        );
        click = () => this.selectMenu(valg.key);
      }

      let element = (
        <ActionButton
          ariaLabel={valg.label}
          onClick={click}
          icon={valg.icon}
          className={this.styles.navMenuchoiceButton}
        >
          {choiceContent}
        </ActionButton>
      );

      if (valg.href) {
        element = this.buidNavMenuLinkChoice(element, valg.href);
      }

      return (
        <li key={valg.key} className={classnames(classes)}>
          {element}
          {dropdown}
        </li>
      );
    });
  }

  render() {
    const menuchoiceList = this.buildNavMenu();

    const showOverlay =
      this.state.activeMenu !== null ? this.styles.overlayShow : '';

    return (
      <div>
        <div
          id={'topStripeOverlay'}
          className={classnames(this.styles.overlay, showOverlay)}
          onClick={() => this.selectMenu(null)}
        />

        <div className={this.styles.container}>
          <div id={'topStripeNav'} className={this.styles.nav}>
            <ul className={classnames(this.styles.navMenu)}>
              {menuchoiceList}
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

export default TopStripe;
