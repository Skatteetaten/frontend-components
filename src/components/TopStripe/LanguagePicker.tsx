import * as React from 'react';
import { TopStripeButton } from './TopStripeButton';
import { TopStripeMenu } from './TopStripeMenu';
import { EnglishFlag, NorwegianFlag, SamiFlag } from './assets';
import { Icon } from '../Icon';
import { UseScreen } from '../utils';
import { getClassNames } from './LanguagePicker.classNames';

export enum LanguageEnum {
  BOKMAAL = 'nb',
  NYNORSK = 'nn',
  ENGLISH = 'en',
  SAMI = 'se',
}

const generateLanguagePickerText = (language: LanguageEnum): string => {
  switch (language) {
    case LanguageEnum.BOKMAAL:
      return 'Bokmål';
    case LanguageEnum.NYNORSK:
      return 'Nynorsk';
    case LanguageEnum.ENGLISH:
      return 'English';
    case LanguageEnum.SAMI:
      return 'Sámigiella';
  }
};

const displayFlag = (language: LanguageEnum): JSX.Element => {
  const altText: string = generateLanguagePickerText(language);

  switch (language) {
    case LanguageEnum.BOKMAAL:
      return <img src={NorwegianFlag} alt={altText} />;
    case LanguageEnum.NYNORSK:
      return <img src={NorwegianFlag} alt={altText} />;
    case LanguageEnum.ENGLISH:
      return <img src={EnglishFlag} alt={altText} />;
    case LanguageEnum.SAMI:
      return <img src={SamiFlag} alt={altText} />;
  }
};

const LanguagePickerButton = ({
  buttonLanguage,
  selectedLanguage,
  setLanguage,
  showOnMobile,
}): JSX.Element => {
  const styles = getClassNames();
  return (
    <TopStripeButton
      onClick={() => setLanguage(buttonLanguage)}
      showOnMobile={showOnMobile}
      className={styles.languageButton}
      role={'menuitem'}
      aria-current={buttonLanguage === selectedLanguage}
    >
      {buttonLanguage === selectedLanguage && (
        <Icon iconName={'Check'} className={styles.checkIcon} />
      )}
      <span className={styles.flag}>{displayFlag(buttonLanguage)}</span>
      <span className={styles.text}>
        {generateLanguagePickerText(buttonLanguage)}
      </span>
    </TopStripeButton>
  );
};

export interface LanguagePickerProps {
  className?: string;
  /**
   * Current language on the page. Valid values are 'nb', 'nn', 'en' and 'se'
   */
  selectedLanguage: LanguageEnum;
  /**
   * Method for setting the language on the page
   */
  setLanguage: any;
  /**
   * If element is displayed in TopStripe on mobile
   * @default true
   */
  showOnMobile?: boolean;
  /**
   * If sami option is included in language picker
   * @default false
   */
  showSami?: boolean;
}

export const LanguagePicker: React.FC<LanguagePickerProps> = (props) => {
  const {
    className,
    selectedLanguage,
    setLanguage,
    showOnMobile = true,
    showSami = false,
  } = props;
  const [languages, setLanguages] = React.useState<LanguageEnum[]>([]);

  React.useEffect(() => {
    showSami
      ? setLanguages([
          LanguageEnum.BOKMAAL,
          LanguageEnum.NYNORSK,
          LanguageEnum.ENGLISH,
          LanguageEnum.SAMI,
        ])
      : setLanguages([
          LanguageEnum.BOKMAAL,
          LanguageEnum.NYNORSK,
          LanguageEnum.ENGLISH,
        ]);
  }, [showSami]);

  const styles = getClassNames();
  const screenSize = UseScreen();
  if (screenSize.sm && !showOnMobile) {
    return null;
  }

  return (
    <>
      <span className={styles.flag}>{displayFlag(selectedLanguage)}</span>
      <TopStripeMenu
        showOnMobile={showOnMobile}
        title={generateLanguagePickerText(selectedLanguage)}
        className={className}
      >
        {languages.map((language) => {
          return (
            <LanguagePickerButton
              buttonLanguage={language}
              selectedLanguage={selectedLanguage}
              setLanguage={setLanguage}
              showOnMobile={showOnMobile}
            />
          );
        })}
      </TopStripeMenu>
    </>
  );
};
