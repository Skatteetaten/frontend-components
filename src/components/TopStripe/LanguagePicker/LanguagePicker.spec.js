import React from 'react';
import { mount } from 'enzyme';
import { LanguagePicker } from '.';
import { TopStripeContext } from '../TopStripe';

function oppsettFullDOM(props) {
  return mount(
    <TopStripeContext.Provider
      value={{
        open: 0,
        setOpen: jest.fn(),
      }}
    >
      <LanguagePicker {...props} />
    </TopStripeContext.Provider>
  );
}

describe('LanguagePicker komponent', () => {
  const setOpenMock = jest.fn();

  it('matcher snapshot', () => {
    const wrapper = oppsettFullDOM({
      selectedLanguage: 'nb',
      setLanguage: jest.fn(),
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('N친r knappen trykkes p친, kalles setOpen', () => {
    const wrapper = mount(
      <TopStripeContext.Provider
        value={{
          open: 0,
          setOpen: setOpenMock,
        }}
      >
        <LanguagePicker />
      </TopStripeContext.Provider>
    );
    const button = wrapper.find('button');
    button.simulate('click');
    wrapper.update();
    expect(setOpenMock).toHaveBeenCalled();
  });

  it.each([
    ['Bokm친l', 'nb'],
    ['Nynorsk', 'nn'],
    ['English', 'en'],
    ['S치megiella', 'se'],
  ])(
    'rendrer med tekst (%p) dersom selectedLanguage er satt til (%p)',
    (selectedLanguageText, selectedLanguage) => {
      const wrapper = oppsettFullDOM({
        selectedLanguage: selectedLanguage,
        setLanguage: jest.fn(),
      });
      const menuButton = wrapper.find('button');
      expect(menuButton.text().includes(selectedLanguageText)).toEqual(true);
    }
  );
});
