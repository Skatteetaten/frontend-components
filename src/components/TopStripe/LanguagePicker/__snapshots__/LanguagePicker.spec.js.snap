// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagePicker komponent matcher snapshot 1`] = `"<button class=\\"topStripeMenuShowChevron-116 topStripeButton-125\\" style=\\"cursor: pointer;\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><div class=\\"topStripeMenuTitle-118\\"><span class=\\"flag-113\\"><img src=\\"NorwegianFlag.svg\\" alt=\\"\\"></span>Bokmål</div><i data-icon-name=\\"MenuDown\\" aria-hidden=\\"true\\" class=\\"topStripeMenuChevronIcon-128\\"></i></button>"`;
